From 8ff326ccdcb3c0c0905eb0e21e86987af63bf507 Mon Sep 17 00:00:00 2001
From: Sai Krishna Potthuri <lakshmi.sai.krishna.potthuri@xilinx.com>
Date: Fri, 18 Sep 2020 13:24:30 +0530
Subject: [PATCH 1601/1851] firmware: xilinx: Updated the reset function
 prototypes

commit 2740c6bf409e16a64217787efa6bc635ce22852f from
https://github.com/Xilinx/linux-xlnx.git

This patch updates the reset function prototypes to make it
inline with the function definitions.
Instead of enum zynqmp_pm_reset, use u32 for reset ID argument.

Addresses-coverity: enumerated type mixed with another type.
Signed-off-by: Sai Krishna Potthuri <lakshmi.sai.krishna.potthuri@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 include/linux/firmware/xlnx-zynqmp.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/linux/firmware/xlnx-zynqmp.h b/include/linux/firmware/xlnx-zynqmp.h
index f8f00a75b11d..a644be954a65 100644
--- a/include/linux/firmware/xlnx-zynqmp.h
+++ b/include/linux/firmware/xlnx-zynqmp.h
@@ -592,9 +592,9 @@ struct zynqmp_eemi_ops {
 	int (*clock_setparent)(u32 clock_id, u32 parent_id);
 	int (*clock_getparent)(u32 clock_id, u32 *parent_id);
 	int (*ioctl)(u32 node_id, u32 ioctl_id, u32 arg1, u32 arg2, u32 *out);
-	int (*reset_assert)(const enum zynqmp_pm_reset reset,
+	int (*reset_assert)(const u32 reset,
 			    const enum zynqmp_pm_reset_action assert_flag);
-	int (*reset_get_status)(const enum zynqmp_pm_reset reset, u32 *status);
+	int (*reset_get_status)(const u32 reset, u32 *status);
 	int (*init_finalize)(void);
 	int (*set_suspend_mode)(u32 mode);
 	int (*request_node)(const u32 node,
-- 
2.31.1

